core/base-service/route.spec.js

```diff
@@ -46,33 +67,41 @@ describe('Route helpers', function () {
       given('/foo/bar/bar.svg').expect(null)
     })
 
-    const namedParams = str =>
-      namedParamsForMatch(captureNames, regex.exec(str))
-    test(namedParams, () => {
-      forCases([
-        given('/foo/bar.bar.bar.svg'),
-        given('/foo/bar.bar.bar.json'),
-      ]).expect({ namedParamA: 'bar.bar.bar' })
+    const params = path =>
+      paramsForPath({ regex, captureNames, ServiceClass }, path)
+    test(params, () => {
+      given('/foo/bar.bar.bar.svg').expect({
+        namedParams: { namedParamA: 'bar.bar.bar' },
+        format: 'svg',
+      })
+      given('/foo/bar.bar.bar.json').expect({
+        namedParams: { namedParamA: 'bar.bar.bar' },
+        format: 'json',
+      })
 
       // This pattern catches bugs related to escaping the extension separator.
-      given('/foo/bar.bar.bar_svg').expect({ namedParamA: 'bar.bar.bar_svg' })
-      given('/foo/bar.bar.bar.zip').expect({ namedParamA: 'bar.bar.bar.zip' })
+      given('/foo/bar.bar.bar_svg').expect({
+        namedParams: { namedParamA: 'bar.bar.bar_svg' },
+        format: 'svg',
+      })
+      given('/foo/bar.bar.bar.zip').expect({
+        namedParams: { namedParamA: 'bar.bar.bar.zip' },
+        format: 'svg',
+      })
     })
   })
 
   context('No named params are declared', function () {
     const { regex, captureNames } = prepareRoute({
       base: 'foo',
-      format: '(?:[^/]+)',
+      format: '(?:[^/]+?)',
```

Comment:
To correctly detect `.svg` and `.json` as intended, this regex needs to be lazy.